### PR TITLE
test: fix the flaky test-tcp-reuseport

### PR DESCRIPTION
**Pull request by panjf2000**
Start connecting to the peers after all threads
to poll for accepting connections.

For #4407

### CODE REVIEW

**Comment by panjf2000** [comment]
It seemed that connecting to the listeners that have not yet started accepting may not get the incoming connections to be distributed across all listening threads. Check out https://github.com/libuv/libuv/actions/runs/9190479225/job/25274784544

File: test/test-tcp-reuseport.c [hunk 1]
```diff
@@ -157,22 +158,36 @@ TEST_IMPL(tcp_reuseport) {
...
+  /* Run event loops of listeners in separate threads. */
+  uv_thread_t thread_loop_id1;
+  uv_thread_t thread_loop_id2;
+  uv_thread_create(&thread_loop_id1, run_event_loop, thread_loop1);
+  uv_thread_create(&thread_loop_id2, run_event_loop, thread_loop2);
+
+  /* Sleep to ensure all threads to poll for accepting connections
+   * before we start to connect. Otherwise the incoming connections
+   * might not be distributed across all listening threads. */
+  uv_sleep(100);
```
Comment (saghul) [inline test/test-tcp-reuseport.c:187]: Perhaps you could use a semaphore to make sure rather than relying on timing?

**Review by saghul — APPROVED** [review]
LGTM with a suggestion

File: test/test-tcp-reuseport.c (hunk 1 above)
Comment (panjf2000) [reply test/test-tcp-reuseport.c:200]: Good point, done.

**Review by saghul — APPROVED** [review]

**Comment by panjf2000** [comment]
Hi @santigimeno, got a minute for this?

**Review by saghul — APPROVED** [review]

**Pull request merged**